extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java

```diff
@@ -170,6 +171,93 @@ public void setIncludeClientId(boolean includeClientId) {
     @ConfigItem
     public Logout logout = new Logout();
 
+    public static enum VerificationMode {
```

Comment:
The only thing I'm not sure of is if we can remove these options and decide on the source of the key material automatically. For instance:

* First, check if the `x5c` header is present. If so, check the signature based on the chain there.
* Otherwise, try the JWKS endpoint from the OP

Wouldn't that work well and help to remove the option entirely?

Reply:
@pedroigor Sure, I can experiment - note in the original use case, the provider of such tokens has no any OIDC endpoints (see the test config) but I think it may work like this:

- if `quarkus.oidc.auth-server-url` is absent then do not fail if `quarkus.oidc.certificate-chain.truststore` is set - as it an implicit message that tokens with `x5c` can be accepted
- if `quarkus.oidc.auth-server-url` is present and `quarkus.oidc.certificate-chain.truststore` is also set, then indeed, treat `x5c` as another fallback after `kid` key id and `x5t` and `x5t#S256` thumbprints found no matching JWK keys

Sounds good, the fewer properties, the better, thanks for the idea